File: yearcheck.js
// current month / year - old month / year
// give number of years past
// 09/19 - 03/15 = 4.5 years
// current time should be updated automatically not hard coded
// static it first then make it dynamic

const readline = require('readline-sync');
const date = new Date();

// Get user input month
let pastMonth = readline.question('Enter a month 1-12: ');

// Get user input year
let pastYear = readline.question('Enter a year - xxxx format: ');

// Populate current month and year
let currYear = date.getFullYear();
let currMonth = date.getMonth() + 1;

// Summing up the months years
let totalMonths = 0;
let totalYears = 0;
let timeElapsed = totalYears + totalMonths;

if (pastMonth < currMonth) {
  totalMonths = (currMonth - pastMonth) / 12;
  //   console.log(totalMonths);
}
if (pastYear < currYear) {
  totalYears = currYear - pastYear;
  //   console.log(totalYears);
} else {
  totalMonths = (pastMonth - currMonth) / 12;
  totalYears = pastYear - currYear;
}
console.log('Total years passed is: ', totalYears + totalMonths);

File: yearcheck.c
// https : //stackoverflow.com/questions/1442116/how-to-get-the-date-and-time-values-in-a-c-program

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// initialize struct
typedef struct tm tm;

int main(void)
{
    time_t now;
    time(&now);

    tm *local = localtime(&now);

    int pastMonth = 0;
    printf("Please a numerical month: ");
    // scanf() will allow us to read what the user types in
    scanf("%i", &pastMonth);
    // printf("%i", pastMonth);

    int currMonth = local->tm_mon + 1;

    // This will be sum of months
    int totalMonths = 0;

    int pastYear = 1992;

    int currYear = local->tm_year + 1900;

    // This will be sum of years
    int totalYears = 0;

    // implement Do While loop instead?
    // do [getint]
    // while (something)

    if (pastMonth < currMonth && pastYear < currYear)
    {
        totalMonths = currMonth - pastMonth;
        totalYears = currYear - pastYear;
    }
    else
    {
        totalMonths = pastMonth - currMonth;
        totalYears = pastYear - currYear;
    }

    printf("Months passed: %i\nYears passed: %i\n", totalMonths, totalYears);
}
